commit 2b30d411dbc6eddfb5b4f9afd5a2c57b6f4dd96c
Author: Takashi Iwai <tiwai@suse.de>
Date:   Tue Nov 4 14:02:40 2014 +0100

    ALSA: pcm: Add xrun_injection proc entry
    
    This patch adds a new proc entry for PCM substreams to inject an
    XRUN.  When a PCM substream is running and any value is written to its
    xrun_injection proc file, the driver triggers XRUN.  This is a useful
    feature for debugging XRUN and error handling code paths.
    
    Note that this entry is enabled only when CONFIG_SND_PCM_XRUN_DEBUG is
    set.
    
    Signed-off-by: Takashi Iwai <tiwai@suse.de>

diff --git a/include/sound/pcm.h b/include/sound/pcm.h
index 29eb09ef2969..0b8daeed0a33 100644
--- a/include/sound/pcm.h
+++ b/include/sound/pcm.h
@@ -416,7 +416,10 @@ struct snd_pcm_substream {
 	struct snd_info_entry *proc_status_entry;
 	struct snd_info_entry *proc_prealloc_entry;
 	struct snd_info_entry *proc_prealloc_max_entry;
+#ifdef CONFIG_SND_PCM_XRUN_DEBUG
+	struct snd_info_entry *proc_xrun_injection_entry;
 #endif
+#endif /* CONFIG_SND_VERBOSE_PROCFS */
 	/* misc flags */
 	unsigned int hw_opened: 1;
 };